packages/core/src/render3/di.ts

```diff
@@ -334,7 +334,15 @@ export function getOrCreateInjectable<T>(
     // If the ID stored here is a function, this is a special object like ElementRef or TemplateRef
     // so just call the factory function to create it.
     if (typeof bloomHash === 'function') {
-      enterDI(lView, tNode);
+      if (flags & InjectFlags.SkipSelf) {
+        return getOrCreateInjectable(
+            tNode.parent, getParentInjectorView(getParentInjectorLocation(tNode, lView), lView),
```

Comment:
@crisbeto I worked on a fix to support @SkipSelf for Injectors and noticed that `tNode.parent` may not be available when we cross component boundary, [see a test here](https://github.com/angular/angular/pull/34839/files#diff-881d964a81b3db8c8cb5311b8e3546ecR644). It looks like this code should be updated to handle this case as well ([similar to this line](https://github.com/angular/angular/pull/34839/files#diff-a29d4642602b40bdabe81990eed99e50R353)).)

Reply:
Hmm, would that be a problem for `ViewContainerRef`, `ChangeDetectorRef` and `ElementRef` though (crossing component boundaries)? A node should always be inside another element or a view container so we'd always resolve the value before we have to cross a component boundary. That being said, I might be missing some context here.

Reply:
I believe @AndrewKushnir is talking about the case where the `child` element requesting `SkipSelf` is a `ViewChild` of `parent` instead of a `ContentChild`. 

```ts
@Component({selector: 'parent', template: '<child></child>' }) 
export class Parent {...}
```

If `child` is a root node of the `parent` template, `tNode.parent` would be null because the `parent` element is in a different view and we can't cross component boundaries with the `tNode` tree (because `tNode.parent` would change with each usage of the component).  So `ViewContainerRef` would not be found.

In this case, we should be falling back to the `T_HOST` saved in the `LView`:

```suggestion
            tNode.parent || lView[T_HOST}, getParentInjectorView(getParentInjectorLocation(tNode, lView), lView),
```

Reply:
One more thing to consider: when crossing component boundaries, we have to respect `@Host` config flags as well. In other words, if there is `@SkipSelf` and `Host` together, the `ViewContainerRef` should not be found if it would cross a component boundary.

Reply:
One more thing to consider: when crossing component boundaries, we have to respect `@Host` config flags as well. In other words, if there are `@SkipSelf` and `@Host` flags used together, the `ViewContainerRef` should not be found if it would cross a component boundary.